auslib/test/admin/views/test_rules.py

```diff
@@ -141,9 +141,22 @@ def testNewRuleWithoutPermission(self):
     def testMissingFields(self):
         # But we still need to pass product, because permission checking
         # is done before what we're testing
-        ret = self._post('/rules', data=dict({'product': 'a'}))
+        post_data = {'product': 'a'}
+        ret = self._post('/rules', data=post_data)
+        self.assertEquals(ret.status_code, 400, "Status Code: %d, Data: %s" % (ret.status_code, ret.data))
+        # Connexion and manual request.json validation methods will preempt and throw 400 error as
+        # soon as the first field invalidates. Only one field may be present in the response.data depending
+        # upon the order of fields sorted in swagger yaml file and in views.
+        self.assertTrue('title' in ret.data, msg=ret.data)
```

Comment:
@tarekziade , I've already tried that. The issue is that using a custom RequestBodyValidator requires me to inherit [this](https://github.com/zalando/connexion/blob/master/connexion/decorators/validation.py#L142) class and modify the validate_schema method. The issue here is that the form data is passed as a json object [here](https://github.com/zalando/connexion/blob/master/connexion/decorators/validation.py#L107) (keys are the form field names and values are the field's values). The json is passed to the jsonschema's schema validator method when validate method ,[here](https://github.com/zalando/connexion/blob/master/connexion/decorators/validation.py#L133) , is invoked. Now this inturn invokes jsonschema's [validate method](https://github.com/Julian/jsonschema/blob/master/jsonschema/validators.py#L128) which validates the input json based on swagger schema and raisesan error as soon as it encounters one. 
Now the tricky part : the error response from jsonschema does not specify the field name in the error.message .  This is an issue for us because as mentioned in example above if i pass invalid value to ,say update_type field (which can have either 'major' or 'minor' field value) then the error message just says something like this : 

> "u'mindosr' is not one of ['major', 'minor']",


Now, this means that the field name isn't specified above.  So, as far as I understood, I will have to modify the jsonschema code rather than using a custom RequestBodyValidator (though it can be used later for some custom validations besides this issue) which itself is a bit arduous and a maintenance nightmare for us. What do you think ? Is there a way to circumvent this issue ?